packages/mobile/src/redux/createMigrate.test.ts

```diff
@@ -0,0 +1,113 @@
+import createMigrate from 'src/redux/createMigrate'
+import { migrations } from 'src/redux/migrations'
+import { _persistConfig } from 'src/redux/store'
+import { vNeg1Schema } from 'test/schemas'
+
+const testMigrations = {
+  0: jest.fn((state) => ({ ...state, mig0: true })),
+  1: jest.fn((state) => ({ ...state, mig1: true })),
+  2: jest.fn((state) => ({ ...state, mig2: true })),
+  3: jest.fn((state) => ({ ...state, mig3: true })),
+}
+
+describe(createMigrate, () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  // This ensures all migrations can be run from the initial state
+  it(`works for v-1 to v${_persistConfig.version}`, async () => {
+    const vNeg1Stub = {
+      ...vNeg1Schema,
+      _persist: { version: -1, rehydrated: false },
+    }
+
+    const migrate = createMigrate(migrations)
+    const migratedSchema = await migrate(vNeg1Stub, _persistConfig.version)
+    expect(typeof migratedSchema).toBe('object')
+  })
+
+  it(`returns undefined if there's no persisted state`, async () => {
+    const migrate = createMigrate(testMigrations)
+    const migratedSchema = await migrate(undefined, 0)
+    expect(testMigrations[0]).not.toHaveBeenCalled()
+    expect(testMigrations[1]).not.toHaveBeenCalled()
+    expect(testMigrations[2]).not.toHaveBeenCalled()
+    expect(testMigrations[3]).not.toHaveBeenCalled()
+    expect(migratedSchema).toBe(undefined)
+  })
+
+  // This is what the original createMigrate from redux-persist does,
+  // but should we throw instead?
```

Comment:
I don't think it's a common enough case for it to matter, is there even a way to downgrade using the app/play store?

Reply:
agreed, I don't know of any way for App Store/Play Store users to downgrade.